src/finch/codegen/numpy_buffer.py

```diff
@@ -36,31 +36,32 @@ def __init__(self, arr: np.ndarray):
         if not arr.flags["C_CONTIGUOUS"]:
             raise ValueError("NumPy array must be C-contiguous")
         self.arr = arr
+        self.arr_ref = [self.arr]
 
     def get_format(self):
         """
         Returns the format of the buffer, which is a NumpyBufferFormat.
         """
-        return NumpyBufferFormat(self.arr.dtype.type)
+        return NumpyBufferFormat(self.arr_ref[0].dtype.type)
 
     def length(self):
-        return self.arr.size
+        return self.arr_ref[0].size
 
     def load(self, index: int):
-        return self.arr[index]
+        return self.arr_ref[0][index]
 
     def store(self, index: int, value):
-        self.arr[index] = value
+        self.arr_ref[0][index] = value
```

Comment:
Instead of messing with the NumpyCBuffer, could you make a numba-focused buffer type like NumbaBuffer to do this transform on?